core-bundle/src/EventListener/BackendPreviewRedirectListener.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener;
+
+use Contao\CoreBundle\Routing\ScopeMatcher;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpKernel\Event\RequestEvent;
+
+/**
+ * @internal
+ */
+class BackendPreviewRedirectListener
+{
+    /**
+     * @var ScopeMatcher
+     */
+    private $scopeMatcher;
+
+    public function __construct(ScopeMatcher $scopeMatcher)
+    {
+        $this->scopeMatcher = $scopeMatcher;
+    }
+
+    public function __invoke(RequestEvent $event): void
+    {
+        if (!$event->isMasterRequest()) {
+            return;
+        }
+
+        $request = $event->getRequest();
+
+        if (!$request->attributes->has('_preview')
+            || true !== $request->attributes->get('_preview')
+            || !$this->scopeMatcher->isBackendRequest($request)
```

Comment:
Would it make sense to check for the frontend instead?
```suggestion
            || $this->scopeMatcher->isFrontendRequest($request)
```

Reply:
You mean in case someone has added their own scope?

Reply:
Yes, or their own route without any scope.

Reply:
Idea © by @Toflar https://github.com/contao/contao/pull/3464#issuecomment-918095113

Reply:
Changed in cc58b21e612a18924c370a9cbc4ab83d68fb5278.